harness/tests/cli/test_user.py

```diff
@@ -47,3 +48,29 @@ def test_user_edit_no_fields(mock_die: mock.MagicMock) -> None:
         mock_die.assert_has_calls(
             [mock.call("No field provided. Use 'det user edit -h' for usage.", exit_code=1)]
         )
+
+
+@mock.patch("getpass.getpass")
+@mock.patch("determined.cli.cli.die")
+def test_login_with_invalid_credentials_error_message(
+    mock_die: mock.MagicMock, mock_getpass: mock.MagicMock
+) -> None:
+    mock_getpass.side_effect = lambda *_: "newpass"
+    with responses.RequestsMock(
```

Comment:
The context manager version of this works, but since you're not doing anything complicated, the easier way to do this is to use the `responses.activate` decorator. It uses the default registry, which should be a little less brittle than the `OrderedRegistry`, anyway.

```python
@mock.patch...
@responses.activate
def test_log...
  ...
  util.expect_get_info()  # take a look at the code here for another example of the pattern with `responses.activate`
  responses.post(...)
```

Reply:
I agree `OrderedRegistry` is not necessary for this test.